charts/sftd/templates/service.yaml

```diff
@@ -0,0 +1,36 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "sftd.fullname" . | trunc 55 | trimSuffix "-" }}-headless
+  labels:
+    {{- include "sftd.labels" . | nindent 4 }}
+spec:
+  clusterIP: None
+  ports:
+    - port: 8585
+      targetPort: sft
+      name: sft
+    - port: 49090
+      targetPort: metrics
+      name: metrics
+  selector:
+    {{- include "sftd.selectorLabels" . | nindent 4 }}
+---
+# NOTE: I was not sure if Ingress could point to a headless service. That's why I added this
```

Comment:
question: what does headless mean?

thought: that sounds like sth that can be tested and that would eventually end up in a definitive decision on whether the second service is necessary or not.

Reply:
A headless service is a service with `ClusterIP: None`.  Every `StatefulSet` needs an associated headless service as Kubernetes uses it to assign each pod a stable network id (https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/). 

For 1 replica the usefulness of that is less clear and it's just a ceremony we need to do for `StatefulSet` to work. But in the follow-up PR adding high availability the headless service allows us to point  the load-balancer to individual pods (e.g. to join an existing call).

https://github.com/wireapp/wire-server-deploy/pull/383/commits/a0830a8ea52bc36363d7b248b5fe144b4e0553f3#diff-15db362c68a8e0f375c147772351b59e6512d29eacfca2fde4a66e9dbeb9650fR20

Headless services don't provide a ClusterIP that does load-balancing; but gives SRV records instead. I wasn't sure if this would work with `nginx-ingress`. I'll just give it a try and see.

Reply:
Apparently ingress-nginx uses the Endpoints object directly instead of going through the `ClusterIP` to avoid an additional network hop through `kube-proxy`. So seems we can indeed remove the extra service: https://github.com/kubernetes/ingress-nginx/issues/257

Will do this now

Reply:
Done: https://github.com/wireapp/wire-server-deploy/pull/382/commits/37db5b8f9c6fa02bdc7f0d63e1fb98ea38c553ca